### PR TITLE
PR for Open-Typer app request

### PR DESCRIPTION
**Pull request by Sussy-OS**
This PR is for adding Open-Typer to pi-apps.

### CODE REVIEW

File: apps/Open-Typer/install-32 [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+status "Creating Menu Button"
+sudo mkdir -p /usr/local/share/applications
+echo "[Desktop Entry]
+Name=Open-Typer
+Comment=An Open Source spiritual successor to the flash game Line Rider
+Icon=$(dirname "$0")/icon-64.png
+Exec=/opt/Open-Typer.AppImage %f
+Type=Application
+Terminal=false
+Categories=Utility;" | sudo tee /usr/local/share/applications/Open-Typer.desktop >/dev/null || error "Failed to create menu button!"
```
Comment (Botspot) [inline apps/Open-Typer/install-32:27]: - Are you sure Utility is the best category for this? I was thinking Education would be the best fit.
- Please update the comment.
- The `%f` argument in the Exec line denotes the ability to open files. Does Open-Typer actually have the ability to open any kind of files?

**Comment by Botspot** [comment]
Any suggestions for which category this would fit best in?

**Comment by adazem009** [comment]
> Any suggestions for which category this would fit best in?

I'm the developer of Open-Typer and I use the `Education` category in all .desktop files. For example: https://github.com/Open-Typer/Open-Typer/blob/master/res/linux-release/usr/share/applications/open-typer-appimage.desktop

File: apps/Open-Typer/install-32 (hunk 1 above)
Comment (Sussy-OS) [reply apps/Open-Typer/install-32:27]: Meant to change Utility, sorry. Did not know how to execute an appimage from a menu entry, and saw both Cura and OpenSCAD use it, so I put it in. You can remove it.

Comment (Sussy-OS) [reply apps/Open-Typer/install-32:27]: Any idea what %U does then? adazem009 thought it was also files.

Comment (Sussy-OS) [reply apps/Open-Typer/install-32]: A quick search told me it was for accessing urls. 
image

**Comment by Sussy-OS** [comment]
Accidentally kept the wrong comment in the menu entry, so just fixed that. All seems well and mergeable now.

**Comment by Botspot** [comment]
> > Any suggestions for which category this would fit best in?
> 
> I'm the developer of Open-Typer and I use the `Education` category in all .desktop files. For example: https://github.com/Open-Typer/Open-Typer/blob/master/res/linux-release/usr/share/applications/open-typer-appimage.desktop

Sorry, I meant pi-apps category. We don't have an education category, because until now everything educational has been about coding, so we've put them all in the Programming category.

I will give this a quick test soon and afterwards will probably have a better idea where it best fits.

**Comment by Sussy-OS** [comment]
> > > Any suggestions for which category this would fit best in?
> > 
> > 
> > I'm the developer of Open-Typer and I use the `Education` category in all .desktop files. For example: https://github.com/Open-Typer/Open-Typer/blob/master/res/linux-release/usr/share/applications/open-typer-appimage.desktop
> 
> Sorry, I meant pi-apps category. We don't have an education category, because until now everything educational has been about coding, so we've put them all in the Programming category.
> 
> I will give this a quick test soon and afterwards will probably have a better idea where it best fits.

Tools possibly? I would still try it though, so you can get a better opinion.

**Comment by theofficialgman** [comment]
please stop doing this https://github.com/Botspot/pi-apps/pull/2421/commits/f65b36d02519bb5abde4719be0f63964e9082679

it dirtys the git history and makes it impossible for our CI to run.
you have multiple files that have incorrect permissions btw (need executable bit set)

File: apps/Open-Typer/install-32 [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+version=5.2.0
+
+#Install Depends
+install_packages libfuse2 || exit 1
```
Comment (theofficialgman) [inline apps/Open-Typer/install-32:6]: don't directly do this.
call `enable_module fuse`. refer to other pi-apps scripts

also I personally prefer one `install` script if the entire script is duplicated between 32/64bit. refer to other pi-app scripts for how to use `case` with the arch variable

**Comment by Sussy-OS** [comment]
> please stop doing this [f65b36d](https://github.com/Botspot/pi-apps/commit/f65b36d02519bb5abde4719be0f63964e9082679) (merge commit). You should use the rebase tool (via git or your favorite git GUI client such as Github Desktop)
> 
> it dirtys the git history and makes it impossible for our CI to run. you have multiple files that have incorrect permissions btw (need executable bit set)

Will remember.

File: apps/Open-Typer/install-32 (hunk 2 above)
Comment (Sussy-OS) [reply apps/Open-Typer/install-32:6]: Alright, I'll do that right now.

Comment (Sussy-OS) [reply apps/Open-Typer/install-32:6]: Pushing now, I hope the permission scopes are correct.

**Comment by Sussy-OS** [comment]
Are there any other issues? I have tested it, and it runs perfectly.

**Comment by theofficialgman** [comment]
sorry going to have to close this. it can't be included with such high GLIBC requirements. that means it can't run on buster and bionic.

@adazem009 we can re-open if you build your appimages on an older distro for compatabiilty. I suggest setting up build chroots or docker containers on ubuntu bionic. keep in mind that github actions (since I see you use it at least partially for CI purposes) can build in a docker container easily with `container: nameofcontainer` at the same level as the `runs-on:` version. eg: `container: ubuntu:18.04`

**Comment by adazem009** [comment]
@theofficialgman I checked that PPA before, but I need at least Qt 6.4, so it's better to cross-compile it on x86 (building it in inside an emulator is definitely not a choice because of GitHub Actions limit).

I'll try to build Open-Typer 5.3 on an older distro (I'll release it in October).

**Comment by adazem009** [comment]
@theofficialgman I think I'll keep the builds on Ubuntu 20.04. Bionic has a too old toolchain and I need at least C++17 for Qt 6. I've been trying to build a gcc cross compiler myself for a few weeks, but even after I'd managed to do it, it didn't pass the "compiler tests" Qt runs before building.

I'll move this to the next minor version and try to build Qt 6 in a PPA instead. It should be easier to build it directly on the target architecture.

**Comment by Botspot** [comment]
Reopening now that we have dropped Buster and Bionic support.

**Comment by theofficialgman** [comment]
@Sussy-OS thanks for the PR, it has been merged
@adazem009 thanks for the development of this application. One thing I will not about it is the displayed keyboard seems useless to me since it can't be interacted with (ubuntu gnome 22.04 x11 session with a device with a builtin touchscreen)

**Comment by adazem009** [comment]
> @Sussy-OS thanks for the PR, it has been merged
> @adazem009 thanks for the development of this application. One thing I will note about it is the displayed keyboard seems useless to me since it can't be interacted with (ubuntu gnome 22.04 x11 session with a device with a builtin touchscreen). I am sure that is intentional, it just looks like on on-screen intractable keyboard to me 

The application is designed specifically for the desktop to help people learn to type on a physical keyboard.
I don't think it would make sense to make it interactable because there are many different on-screen keyboards people use.

However, I could show the native keyboard on touch screen devices instead.

**Pull request merged**